src/family_name_classifier.py

```diff
@@ -0,0 +1,315 @@
+import builtins
+
+import pandas as pd
+import numpy as np
+import itertools
+import unicodedata
+import re
+import collections
+import random
+import sklearn.feature_extraction
+import sklearn.linear_model
+import sklearn.cross_validation
+import sklearn.externals.joblib
+random.seed(0)
+
+
+class FamilyNameClassifier:
+    IGNORE_LIST = {'de', 'da', 'do', 'des', 'das', 'dos', 'e', 'dr', 'sobrinho', 'junior', 'neto', 'jr', 'filho'}
+    NOT_ACCEPTABLE_CHARACTERS = re.compile(r'[^a-z ]')
```

Comment:
Do we need to remove `ç` and `é` and so on, or just punctuation, numbers? I suggested below we could convert `ç` to `c`, `é` to `e` and so on — but not sure if that means a better classifier… what do you think about it?